### ISSUE TITLE
First invitations and thoughts about the Organization

### ISSUE THREAD

**Issue by tunnckoCore** [regexhq/regexhq#2]
### Mentioning, lol

@jonschlinkert @doowb @kevva @contra @dperini @jonathanong @juliangruber @sindresorhus @slevithan @mathiasbynens @tj @coreh @dougwilson @pillarjs @assemble

@puzrin , it seems you dont like very much regexps, but you maybe can agree with that they do good things in some cases. Your `no regex` approach might be useful here.
### Goal, short desc

> Centralized place for community-driven categorized/modularized/performanced/small/medium collections of RegEx patterns, groups and tools that can make our life better.

**I'm workin' on draft yet, but if you have some thoughts/ideas inspired by the description, feel free to share them! Feel free to mention another people, share another useful modules, patterns, etc!**
### Useful refs / tools (already added to the wiki page)
- [ ] http://ostermiller.org/findcomment.html
- [ ] http://stackoverflow.com/a/3524574
- [x] http://www.gethifi.com/tools/regex
- [x] http://www.regexr.com
### Some points / Goals
- language agnostic as possible
- core module: extensible
  - all modules to be some sort of plugin? Or like Underscore approach? Categorized collections? 
- core module: easy, flexible api
- core module: CLI
- browser compatibility
- unit tests
- performance tests
- RegExps Spec.
- ports for another languages - Go, C#, Python, Ruby, PHP, etc.
  - JS to be handcrafted JS, not ugly compiled JS from Coffee-Script
### Some packages
- on top of [slevithan/XRegExp](https://github.com/slevithan/XRegExp) ?
- [wearefractal/BetterRegExp](https://github.com/wearefractal/BetterRegExp)
- [wearefractal/xemplar](https://github.com/wearefractal/xemplar)
- [component/regexps](https://github.com/component/regexps)
- [kevva/url-regex](https://github.com/kevva/url-regex)
- [webmodules/urlregexp](https://github.com/webmodules/urlregexp)
- [strip-comments's regexps](https://github.com/jonschlinkert/strip-comments/blob/master/index.js#L10-L13)
- [jonschlinkert/gfm-code-blocks-regex](https://github.com/jonschlinkert/gfm-code-blocks-regex)
- [jonschlinkert/code-context](https://github.com/jonschlinkert/code-context) / [tunnckoCore/js-code-context](https://github.com/tunnckoCore/js-code-context)
- [pillarjs/path-to-regexp](https://github.com/pillarjs/path-to-regexp)
- [jonschlinkert's regexps](https://github.com/search?q=user%3Ajonschlinkert+regex)
- [sindresorhus's regexps](https://github.com/search?q=user%3Asindresorhus+regex)
- [kevva's regexps](https://github.com/search?q=user%3Akevva+regex)
- **Be social, share more**
### Be social, suggest ideas, share thoughts about
- [x] organizing the Org
- [x] repo names, conventions (#6)  ~~\- I think `regexps.js`, `regexps.go` and etc for the main module~~
- how you organize your orgs teams
- moving existing repos/modules to $this?
- etc ...

**Comment by yocontra** [comment]
I'll rewrite the BetterRegExp module to suit our needs - if anyone has any new ideas let me know

**Comment by jonathanong** [comment]
wanted this for a while, but never really found any super-useful regexps. i can slowly move some https://github.com/component regexps here

**Comment by yocontra** [comment]
I think for module organization it makes sense to bundle by purpose

Example:

``` js
// phones is an object with multiple regexps
var phones = require('phone-number-regexp');

// more specific regexps can be required or grabbed off the main module exports
var canadian = require('phone-number-regexp/canada');
```

Not sure which suffix on the module name makes more sense, regex or regexp.

**Comment by jonathanong** [comment]
`path-to-regexp` won't be moving though.

**Comment by jonschlinkert** [comment]
> wanted this for a while, but never really found any super-useful regexps

Yep, agreed. I can move some of ours/mine over here too.

**Comment by jonschlinkert** [comment]
> Not sure which suffix on the module name makes more sense

I tend to use them both in round-robin just hedge my bets lol. jk but really I don't really have a preference.  regexp seems like the more commonly used abbrev, but regex seems more natural to say... just my 2c.

> I think for module organization it makes sense to bundle by purpose

agreed with that too. e.g. https://github.com/jonschlinkert/author-regex/blob/master/index.js

**Comment by jonschlinkert** [comment]
@tunnckoCore you want to add those of us who have given consent as owners, so we can start doing stuff? I like this idea, it would be nice to finally see some high quality, curated regexps come out of this.

**Comment by kevva** [comment]
> Not sure which suffix on the module name makes more sense

`regex` is nicer imo.

**Comment by dougwilson** [comment]
I would like a module that brings Perl 5's regular express grammar to JavaScript in some way and made it easier to compose multiple regular expressions into a larger one.

**Comment by tunnckoCore** [comment]
**Great! Let's do it!** :)

> @dougwilson: I would like a module that brings Perl 5's regular express grammar to JavaScript in some way and made it easier to compose multiple regular expressions into a larger one.

Sounds good.

> @contra: I think for module organization it makes sense to bundle by purpose

Yea, agreed.

> @contra: Not sure which suffix on the module name makes more sense

Or prefix? If prefix it be better to be as org - `regexp-phone-number`, `regexp-email`, `regexp-url`, `regexp-skype`? If suffix `regex` sounds better and more natural as @jonschlinkert said.
Or as in BetterRegExp - `information-regex`, `security-regex` and so on.

Also I think for **browser compatibility** (added in list above), because in fact, its just a RegEx.

Im not familiar with module patterns, but I like this ones here https://github.com/tunnckoCore/blankr/issues/7 , more this one below

``` js
(function(global) {
  'use strict';

  // body here

  if (typeof define === 'function' && define.amd) {
    define(className);
  } else if (typeof exports === 'object') {
    module.exports = className;
  } else {
    global.className = className;
  }
})(this);
```

**Comment by tunnckoCore** [comment]
As `regexp` have `regexps`, how is the `regex` multiple? LOL! `regexs` sounds ugly.
So I think it would be better everywhere to be `regexp` ?

**Comment by tunnckoCore** [comment]
Maybe something like this, follow me
- author-regex
- skype-regex
- url-regex
- email-regex
- comment-regexps
- security-regexps
- number-regexps

When it will be for some like category/collection and will hold more than 1-2 regular expressions to be with `regexps` suffix, when its single-doing thing with `regex` suffix

**Comment by johno** [comment]
This would be pretty awesome resource to have. I'd be willing to take on any Ruby ports, @tunnckoCore.

**Comment by sindresorhus** [comment]
I prefer `regex` too.

**Comment by tunnckoCore** [comment]
@johnotander, great!

> you want to add those of us who have given consent as owners, so we can start doing stuff?

@jonschlinkert, @jonathanong, you are invited ;] @contra already accepts. Anyone else?

**Comment by kevva** [comment]
Regarding module patterns, just use keep using CommonJS or provide a `browser.js` file generated with browserify.

**Comment by tunnckoCore** [comment]
@kevva Lol, I think its too much whole browserify for some like [jonschlinkert/author-regex/index.js](https://github.com/jonschlinkert/author-regex/blob/master/index.js)

**Comment by jonschlinkert** [comment]
@tunnckoCore it's about compatibility in other environments - no matter the size of the module

**Comment by tj** [comment]
personally I'd probably go with a language-agnostic approach and work npm/node on top, I think there have been a lot of regular expression package manager things in the past but the only ones I remember were full-blown social networks for regexps haha

**Comment by jonschlinkert** [comment]
@tunnckoCore yeah I think just send invites to everyone who has commented here so far. @kevva, @johnotander, @dougwilson and @sindresorhus

**Comment by jonschlinkert** [comment]
oh, and now @tj if he wants.

**Comment by tunnckoCore** [comment]
> personally I'd probably go with a language-agnostic approach and work npm/node on top

Yea, really what I want/think to be this org. You can help for Go porting, if there have some differences in regex engine, idk.

**Comment by johno** [comment]
I think @tj might be right about staying as language agnostic as possible, with a primary package manager/distribution mechanism (like npm). It keeps things simple.

I also prefer `regex`, and the plural would likely be `regexes`? Coincidentally, `regexes` appears to be an available org name, too. That could be an option.

**Comment by tunnckoCore** [comment]
@johnotander 

> Coincidentally, regexes appears to be an available org name, too.

Lol, missed this thing and `regexes` form. Lets rename it than?
- `language agnostic as possible` added to the list

**Comment by tj** [comment]
Yeah if it's more or less just a list of regexps we can auto-generate packages for them, or have packages that pull from this source remotely, something like that. Doubt it makes sense to have more than one repo for them, similar to homebrew's.

**Comment by dougwilson** [comment]
> Yeah if it's more or less just a list of regexps we can auto-generate packages for them, or have packages that pull from this source remotely, something like that

What language of RegExp would the list be in? Different runtimes have different regular express syntax. Should it just be the JavaScript subset of regular express features?

**Comment by tomByrer** [comment]
:pager: http://www.regexr.com 's author @gskinner 
I've been thinking about a repo also; there are alot of good examples (or at least good starts) in RegExr's 'Community' folder.  Fun to explore, but tricky to find what I need.  I'd rather have a YAML | JSON file I can scan.

**Comment by jonschlinkert** [comment]
IMHO, we have the org/name perhaps we can just mark that item complete. On the other hand, `regular-expressions` is available if no one cares about a few extra chars. 

regarding form

> just a list of regexps we can auto-generate packages for them

love this idea. this is as good of a use-case for auto-generating packages as any I've seen

@tunnckoCore did you get an invite out to @tomByrer?

**Comment by gskinner** [comment]
Since I got copied into this discussion, I may as well comment...

I'd be happy to provide access to the RegExr community library if it helps. I'd also be happy to contribute the odd pattern as I stumble across (or write) ones that seem appropriate.

Also, if this initiative is successful, it seems like something we'd love to utilize in RegExr - I could definitely see including a categorized list of JS-friendly patterns in the app, with attribution and a link back to the project page. We've also been looking at making an embeddable version of the app, so maybe that would be useful on your project page (and it would give me an excuse to actually carve out time to do it).

**Comment by tunnckoCore** [comment]
> Should it just be the JavaScript subset of regular express features?

Language agnostic as possible. I think differences in RegEx between languages are not so much?
I can help in PHP and JS, someone in Go, another in Ruby and so on. :)

> On the other hand, regular-expressions is available if no one cares about a few extra chars.

@jonschlinkert exactly.

@tomByrer invited
@gskinner invited

**Comment by tunnckoCore** [comment]
Added section 

### Useful refs / tools
- http://ostermiller.org/findcomment.html
- http://stackoverflow.com/a/3524574
- http://www.gethifi.com/tools/regex
- http://www.regexr.com
